MapboxNavigation/CarPlaySearchManager.swift

```diff
@@ -0,0 +1,46 @@
+#if canImport(CarPlay)
+import CarPlay
+import MapboxDirections
+
+@available(iOS 12.0, *)
+public protocol CarPlaySearchManagerDelegate: class {
+    func calculateRoute(to toWaypoint: Waypoint, completionHandler: @escaping () -> Void)
+    
+    func resetPanButtons(_ mapTemplate: CPMapTemplate)
+    
+    func pushTemplate(_ template: CPTemplate, animated: Bool)
+    
+    func popTemplate(animated: Bool)
+}
+
+@available(iOS 12.0, *)
+@objc(MBCarPlaySearchManager)
+public class CarPlaySearchManager: NSObject {
```

Comment:
Can you speak to why this functionality needs to be in a separate manager class rather than integrated into CarPlayManager and CarPlayManagerDelegate?

Reply:
This could be an incremental step toward moving all search-related functionality into a different framework (like MapboxGeocoder for example). 

Personally I'd like CarPlayManager to not need to know about search at all if we can avoid it. The CarPlayManager should be about lifecycle events and basic plumbing, at least that's how we were originally thinking about it.

Reply:
@1ec5 Yes, we hope this will address the issue raised here, https://github.com/mapbox/mapbox-navigation-ios/issues/1744, with this solution. The separation of concerns for the CarPlayManager and the CarPlaySearchManager has been highlighted [here](https://github.com/mapbox/mapbox-navigation-ios/issues/1839).